tools/cdk-build-tools/bin/cdk-build.ts

```diff
@@ -4,11 +4,28 @@ import { shell } from '../lib/os';
 import { cdkBuildOptions } from '../lib/package-info';
 import { Timers } from '../lib/timer';
 
+interface Arguments extends yargs.Arguments {
+    force?: boolean;
+    jsii?: string;
+    tsc?: string;
+}
+
 async function main() {
-    const args = yargs
+    const args: Arguments = yargs
+        .env('CDK_BUILD')
         .usage('Usage: cdk-build')
         .option('force', { type: 'boolean', alias: 'f', desc: 'Force a rebuild' })
-        .argv;
+        .option('jsii', {
+            type: 'string',
+            desc: 'Specify a different jsii executable',
```

Comment:
"different" => "custom"/"non-default"

Reply:
P.S. Might be much more useful to allow setting those via env variables (`CDK_BUILD_JSII=xxx`) - I think that should be sufficient for the use cases you are after and less messy on the command line surface area